historical_surveillance/models.py

```diff
@@ -0,0 +1,83 @@
+                    ('secondary school', 'Secondary School'),
+                    ('other', 'Other')]
+
+SEX_CHOICES = [('male', 'Male'),
+               ('female', 'Female')]
+
+
+# TO-DO
+# Enforce Foreign Key constraint on the schools and districts
+
+class District(models.Model):
+    created_at = models.DateField(auto_now_add=True)
+    created_by = models.CharField(max_length=255)
+    District_Code = models.CharField(max_length=50, blank=True)
+    District_Name = models.CharField(max_length=50, choices=DISTRICT_CHOICES)
+    # updated_at = models.DateField(default=timezone.now())
+    updated_at = models.DateField(auto_now_add=True)
+    updated_by = models.CharField(max_length=255, blank=True)
+
+    def __str__(self):
+        return str(self.District_Name)
+
+
+class School(models.Model):
+    objects = None
+    created_at = models.DateField(auto_now_add=True)
+    created_by = models.CharField(max_length=255)
+    School_Code = models.CharField(max_length=50, blank=True)
+    School_Name = models.CharField(max_length=50)
+    District_Name = models.CharField(max_length=20, choices=DISTRICT_CHOICES)
+    category_of_school = models.CharField(max_length=20, choices=CATEGORY_CHOICES)
+    updated_at = models.DateField(auto_now_add=True)
+    updated_by = models.CharField(max_length=255, blank=True)
+
+    def __str__(self):
+        return "Saved"
+
+
+class AggregateEnrollment(models.Model):
+    objects = None
+    created_at = models.DateField(auto_now_add=True)
+    created_by = models.CharField(max_length=255)
+    name_of_school = models.CharField(max_length=55)
+    academic_year = models.CharField(max_length=20)
+    category_of_school = models.CharField(max_length=20, choices=CATEGORY_CHOICES)
+    district_of_school = models.CharField(max_length=20, choices=DISTRICT_CHOICES)
+    capacity_of_school = models.IntegerField()
+    total_enrollment = models.IntegerField()
+    updated_at = models.DateField(auto_now_add=True)
+    updated_by = models.CharField(max_length=255)
+
+    # def save(self, *args, **kwargs):
+    # self.surplus = self.capacity_of_school - self.total_enrollment
+    # super().save(*args, **kwargs)
+
+    def __str__(self):
+        return "Saved"
+
+
+class Enrollment(models.Model):
+    created_at = models.DateField(auto_now_add=True)
+    created_by = models.CharField(max_length=255)
+    school = models.CharField(max_length=50)
+    year = models.CharField(max_length=20)
+    category_of_school = models.CharField(max_length=20, choices=CATEGORY_CHOICES, default=None)
+    district = models.CharField(max_length=20, choices=DISTRICT_CHOICES, default='Select')
+    grade = models.CharField(max_length=20, choices=GRADE_CHOICES, default=None)
+    enrollment = models.IntegerField(null=True, blank=True)
+    sex = models.CharField(max_length=20, null=True, choices=SEX_CHOICES)
+    updated_at = models.DateField(auto_now_add=True)
+    updated_by = models.CharField(max_length=255)
```

Comment:
By default since we are using Django forms - it will not set the updated_by by default. It'd be our responsibility to grab the user in the view that creates, updates and deletes and update this field. I don't see that happening now in the views - which makes me feel like this is useless field right now. 

https://stackoverflow.com/questions/6409541/django-forms-want-to-auto-save-user-client-and-datetime

```
if form.is_valid():
        note = form.save(commit=False)
        note.user = request.user
        note.client = client
        note.save()
```

Where we'd do something like this in the view 
```
if form.is_valid():
        enrollment = form.save(commit=False)
        enrollment.updated_by = request.user
        enrollment.save()
```

Reply:
Yes. I left that there for when the authentication is added, it fills in the name of the logged-in user there in the non-editable form.